test/data.jl

```diff
@@ -945,7 +951,17 @@ end
             )
             resolve_switches!(data)
             @test sort(collect(keys(data["bus"]))) == ["1", "10"]
+            @test data["bus"]["1"]["bus_type"] == 3
         end
         Memento.setlevel!(TESTLOG, level)
     end
+    @testset "Test switches case5_sw 2->1 with 2 inactive" begin
+        # Switch merges 2 -> 1
+        data = PowerModels.parse_file("../test/data/matpower/case5_sw.m")
+        data["bus"]["2"]["bus_type"] = 4
+        @test sort(collect(keys(data["bus"]))) == ["1", "10", "2", "3", "4"]
+        resolve_switches!(data)
+        @test data["bus"]["1"]["bus_type"] == 2
+        @test data["bus"]["2"]["bus_type"] == 4
```

Comment:
I'm not sure what's supposed to happen here. With `propagate_topology_status!` we end up with a bus 2 of type 4. If we don't use `propagate_topology_status!`, bus 2 doesn't appear.

Reply:
Add a test that if the 2-3 switch is activated, we don't merge 3-1 via the inactive 2 bus.

Reply:
Add a test with a cycle between buses via switches.